ducktape/services/service.py

```diff
@@ -52,21 +60,71 @@ def start(self):
                     "Service: %s, node.account: %s" % (self.__class__.__name__, str(node.account)))
             node.account.set_logger(self.logger)
 
-            self.logger.debug("Forcibly cleaning node %d on %s", idx, node.account.hostname)
-            self._clean_node(node)
+    def start(self):
+        """Start the service on all nodes."""
+        self.logger.info("Starting service: %s" % self.__class__.__name__)
+
+        self.logger.debug("Killing processes and attempting to clean up before starting nodes.")
+        for node in self.nodes:
+            # Added precaution - kill running processes
+            self._kill_running_processes(node)
+
+            self.stop_node(node)
+            self.clean_node(node)
+
+        self.logger.info("Starting nodes for %s " % self.__class__.__name__)
+        for node in self.nodes:
+            self.logger.debug("Starting node %d on %s" % (self.idx(node), node.account.hostname))
+            self.start_node(node)
+            self.wait_until_alive(node)
+
+    def start_node(self, node):
+        """Start service process(es) on the given node."""
+        raise NotImplementedError("Subclasses must implement clean_node.")
+
+    def wait_until_alive(self, node):
```

Comment:
Can you explain why this should be separate from start_node? The only use of it I see in this or the muckrake patch is during start/restart, but since restart presumably uses stop/start except in very unusual cases, wouldn't this all be simpler without wait_until_alive?

That said, I think it could be useful for a subclass like DaemonService to separate out start_process and wait_until_alive since since start_process might always be a different command for subclasses but wait_until_alive might follow a common pattern (e.g. checking pid liveness, looking for something in the log, etc).